src/NuGetPackageVerifier/Rules/AssemblyIsBuiltInReleaseConfigurationRule.cs

```diff
@@ -1,45 +1,83 @@
+                    extension.Equals(".exe", StringComparison.OrdinalIgnoreCase))
+                {
+                    try
+                    {
+                        using (var packageFileStream = context.PackageReader.GetStream(currentFile))
+                        using (var stream = new MemoryStream())
+                        {
+                            packageFileStream.CopyTo(stream);
+                            stream.Position = 0;
+                            using (var peReader = new PEReader(stream))
+                            {
+                                var reader = peReader.GetMetadataReader();
+                                foreach (var handle in reader.CustomAttributes)
+                                {
+                                    var customAttribute = reader.GetCustomAttribute(handle);
+                                    var ctor = reader.GetMemberReference((MemberReferenceHandle)customAttribute.Constructor);
+                                    var type = reader.GetTypeReference((TypeReferenceHandle)ctor.Parent);
+                                    var typeName = reader.GetString(type.Name);
+                                    if (string.Equals(typeof(DebuggableAttribute).Name, typeName, StringComparison.Ordinal))
+                                    {
+                                        var attribute = customAttribute.DecodeValue(NullProvider.Instance);
+
+                                        var debuggingMode = (DebuggableAttribute.DebuggingModes)attribute.FixedArguments.Single().Value;
+                                        if (debuggingMode.HasFlag(DebuggableAttribute.DebuggingModes.Default) ||
+                                            debuggingMode.HasFlag(DebuggableAttribute.DebuggingModes.DisableOptimizations))
+                                        {
+                                            issues.Add(PackageIssueFactory.AssemblyHasIncorrectBuildConfiguration(currentFile));
+                                        };
+
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        // This assembly is not managed.
+                    }
+                }
             }
+
+            return issues;
         }
 
-        private static bool HasReleaseConfiguration(Collection<CustomAttribute> assemblyAttributes)
+        private class NullProvider : ICustomAttributeTypeProvider<object>
         {
-            var foundAttr = assemblyAttributes.SingleOrDefault(attr => attr.AttributeType.FullName == typeof(DebuggableAttribute).FullName);
-            if (foundAttr == null)
-            {
-                return false;
-            }
+            public static NullProvider Instance => new NullProvider();
```

Comment:
`public static NullProvider Instance { get; } = new NullProvider();` if you actually want this to be a singleton